### PR TITLE
refactor(api): enforce strict typing of opentrons.protocol_api module

### PR DESCRIPTION
**Pull request by mcous**
## Overview

Last thing I have from tech-debt-Friday: a PR that enforces mypy's strict mode for the contents of `opentrons.protocol_api`

## Changelog

- refactor(api): enforce strict typing of opentrons.protocol_api module

## Review requests

- Most change are type-only changes
- Code removals (see comments below) are safe 

## Risk assessment

Low

### CODE REVIEW

File: api/src/opentrons/protocols/geometry/deck_item.py [hunk 1]
```diff
@@ -2,22 +2,17 @@
 
 
 class DeckItem(abc.ABC):
-    @property  # type: ignore
-    @abc.abstractmethod
-    def highest_z(self):
-        pass
-
-    @highest_z.setter  # type: ignore
```
Comment (mcous) [inline api/src/opentrons/protocols/geometry/deck_item.py:10]: Highest Z setter no longer used; doesn't make sense in a LPC world

File: api/src/opentrons/protocols/context/protocol_api/protocol_context.py [hunk 2]
```diff
@@ -75,19 +74,6 @@ def __init__(
         self._last_mount: Optional[types.Mount] = None
         self._loaded_modules: Set["AbstractModule"] = set()
 
-    @classmethod
-    def build_using(cls, protocol: Protocol, *args, **kwargs):
```
Comment (mcous) [inline api/src/opentrons/protocols/context/protocol_api/protocol_context.py:77]: `ProtocolContextImplementation.build_using` no longer used; the constructor is a better choice

File: api/src/opentrons/protocol_api/module_contexts.py [hunk 3]
```diff
@@ -627,100 +634,100 @@ def execute_profile(
...
-    def block_target_temperature(self):
+    def block_target_temperature(self) -> Optional[float]:
         """Target temperature in degrees C"""
         return self._module.target
 
-    @property  # type: ignore
+    @property  # type: ignore[misc]
     @requires_version(2, 0)
-    def lid_temperature(self):
+    def lid_temperature(self) -> Optional[float]:
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_api/module_contexts.py:704]: Concerning that these all have to be `Optional`. Looks like the lack of typing in this module was covering up some typing weirdness in the underlying hardware API?

**Review by SyntaxColoring — APPROVED** [review]
Thank you!

**Comment by SyntaxColoring** [comment]
Looks like G-code tests are failing on this, for some reason? Weird.

File: api/src/opentrons/protocol_api/module_contexts.py (hunk 3 above)
Comment (mcous) [reply api/src/opentrons/protocol_api/module_contexts.py:704]: Yeah agree

**Comment by mcous** [comment]
> Looks like G-code tests are failing on this, for some reason? Weird.

They were failing because... I made a non-type change that broke the code! Reverted that change, G-Code tests are passing

**Review by SyntaxColoring — APPROVED** [review]

**Pull request merged**